platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/verification/serializer/DependencyVerificationsXmlWriter.java

```diff
@@ -172,9 +172,13 @@ private void writIgnoredKeys(DependencyVerificationConfiguration configuration)
         Set<IgnoredKey> ignoredKeys = configuration.getIgnoredKeys();
         if (!ignoredKeys.isEmpty()) {
             writer.startElement(IGNORED_KEYS);
-            for (IgnoredKey ignoredKey : ignoredKeys) {
-                writeIgnoredKey(ignoredKey);
-            }
+            ignoredKeys.stream().sorted().forEach(ignoredKey -> {
+                try {
+                    writeIgnoredKey(ignoredKey);
+                } catch (IOException ex) {
+                    throw new UncheckedIOException(ex);
+                }
+            });
```

Comment:
```suggestion
            List<IgnoredKey> sortedIgnoredKeys = ignoredKeys.stream().sorted().collect(Collectors.toList());
            for (IgnoredKey ignoredKey : sortedIgnoredKeys) {
                    writeIgnoredKey(ignoredKey);
            }
```

🤔 I think instead of potentially changing the type of exception thrown from this method, it's better to just use non-stream iteration and let the original exception propogate.

Reply:
@tresat I agree in general, but it is the same pattens as used in the methods above, e.g. in `writeGroupedTrustedKey()`. I would rather do a refactoring in the follow up PR if needed

Reply:
Sounds good.